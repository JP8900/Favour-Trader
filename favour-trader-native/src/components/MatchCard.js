import React, {Component} from 'react';
import {StyleSheet, View, Text, Button, } from 'react-native';

class MatchCard extends React.Component{

    render() {
        const { match } = this.props;
        return (
            <View key={match._id} style={styles.matches}>
                <Text style={styles.email} >
                    {match.email}
                </Text>
                <Button title={'view profile'} />
            </View>
        )
    }
}


const styles = StyleSheet.create({
    matches: {
        borderColor: 'black',
        borderWidth: 0.5,
        padding: 2,
        marginBottom: 15,
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 3
        },
        shadowRadius: 5,
        shadowOpacity: 1.0
    },
    email: {
        fontSize: 15,
        marginBottom: 10
    }
});

export default MatchCard;