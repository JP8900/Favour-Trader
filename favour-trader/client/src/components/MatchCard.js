import React, {Component} from 'react';
import {
    Card, CardText, CardBody, Col,
    CardTitle, CardSubtitle, Button
} from 'reactstrap';

class MatchCard extends Component {
    render() {
        const {user, reveal} = this.props;
        return (
            <Col sm="4" className={'pb-3'}>
                <Card>
                    <CardBody>
                        <CardTitle>{user.name.first}</CardTitle>
                        <CardSubtitle>{user.email}</CardSubtitle>
                        <Button size={'sm'} color={'primary'} onClick={reveal}>View Profile</Button>
                    </CardBody>
                </Card>
            </Col>
        );
    }
}

export default MatchCard;