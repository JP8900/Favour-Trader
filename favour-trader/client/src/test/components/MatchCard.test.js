import React from 'react';
import renderer from 'react-test-renderer';
import MatchCard from '../../components/MatchCard.js';

it('renders correctly', () => {
  const user = {
    name: {
      first: 'Jane',
      last: 'Doe',
    },
    email: 'jane.doe@example.com',
  };
  const tree = renderer
    .create(<MatchCard user={user} reveal={() => true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});