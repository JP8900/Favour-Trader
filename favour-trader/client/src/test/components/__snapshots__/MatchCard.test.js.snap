// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pb-3 col col-sm-4"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Jane D.
      </h5>
      <h6
        className="card-subtitle"
      >
        jane.doe@example.com
      </h6>
      <button
        className="btn btn-primary btn-sm"
        onClick={[Function]}
        type="button"
      >
        View Profile
      </button>
    </div>
  </div>
</div>
`;
